src/cascadia/TerminalCore/terminalrenderdata.cpp

```diff
@@ -131,6 +131,30 @@ const std::wstring Microsoft::Terminal::Core::Terminal::GetHyperlinkCustomId(uin
     return _buffer->GetCustomIdFromId(id);
 }
 
+// Method Description:
+// - Gets the regex pattern id of a location
+// Arguments:
+// - The location
+// Return value:
+// - The pattern ID of the location
+const size_t Microsoft::Terminal::Core::Terminal::GetPatternId(const COORD location) const noexcept
+{
+    // Convert the location into its 1-d location because the interval tree stores locations that way
+    const auto absLoc = (_buffer->GetRowByOffset(0).size() * location.Y) + location.X;

```

Comment:
Do we want to maybe add a helper to TextBuffer for the `(_buffer->GetRowByOffset(0).size() * location.Y) + location.X;` calculation? Seems like we use that in a decent number of places

Reply:
Ah, I did wonder if we should have buffer helper functions that accept 1-d coordinates - but if we end up switching the interval tree to hold `til::point`s instead I don't know what the use of those helpers would be.

Reply:
The interval tree now holds `til::point`s so I'm going to resolve this